subprojects/docs/src/samples/userguide/dependencyManagement/workingWithDependencies/accessMetadataArtifact/build.gradle

```diff
@@ -13,16 +13,17 @@ task printGuavaMetadata {
     dependsOn configurations.compileClasspath
 
     doLast {
-        def resolutionResult = configurations.compileClasspath.incoming.resolutionResult
-        def componentIds = resolutionResult.allDependencies.collect { it.selected.id }
+        ResolutionResult resolutionResult = configurations.compileClasspath.incoming.resolutionResult
+        List<ComponentIdentifier> componentIds = resolutionResult.allDependencies.collect { it.selected.id }
 
-        def result = dependencies.createArtifactResolutionQuery().forComponents(componentIds)
+        ArtifactResolutionQuery query = dependencies.createArtifactResolutionQuery()
+            .forComponents(componentIds)
```

Comment:
❌ This is overcomplicated and inefficient if you really only want the pom for guava, you can just pass the String `com.google:guava:18.0` here. No need to first get the dependency graph (which is very expensive).

Reply:
Can you clarify? I don't understand what method you are referring to when you say "pass in the String". The method `fromComponents` only takes one or many `ComponentIdentifier`s. If we want to use a `ComponentIdentifier` then we'd have to create a new one via an internal API.

Reply:
fromComponents was changed to take Strings. Unless we removed that again @jjohannes?

Reply:
I don't see that method in https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/query/ArtifactResolutionQuery.html or on `master`.

Reply:
I don't know anything about this. It does not look like any change made it to master. Do you remember in which context we wanted to change this @oehme?

Reply:
The method is called `forModule`: https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/query/ArtifactResolutionQuery.html#forModule-java.lang.String-java.lang.String-java.lang.String-